src/action/listdir.rs

```diff
@@ -323,15 +320,16 @@ impl super::Request for Request {
         let path_type = pathspec.pathtype
             .ok_or(missing("path type"))?;
         if path_type != PathType::Os as i32 {
-            return Err(ParseError::unsupported_value(String::from("path type"),
-                                                     path_type.to_string()));
+            return Err(session::ParseError::from(
+                ParseError::unsupported_value(String::from("path type"),
+                                              path_type.to_string())));
```

Comment:
You can simplify `session::ParseError::from(something)` with `something.into()`.

Reply:
Moreover, once you apply the suggestion above about generalization of `unsupported_value`, you can simplify the `String::from("path type")` with just `"path_type"`.